### ISSUE TITLE
Ine

### ISSUE THREAD

**Issue by tkimnguyen** [plone/mockup#588]

**Comment by domruf** [comment]
with 247b92228640bd0d184dc4a3abf4a092d2126ffe the dialog switches to Internal Link again

**Comment by frapell** [comment]
Maybe for the first point, when an image is selected / dropped, the Upload button could turn Green or something?

@domruf That means that when the upload finishes, the tab is switched back to the "internal image" and also the image is already selected?

**Comment by domruf** [comment]
@frapell yes.
vangheem originally did this back in august 4ae86f0face4b1ea8998c38a70b430e6ea95fdf5
But it didn't work anymore.
With my fix in 247b922 it works again.

**Comment by domruf** [comment]
Is there a reason not to automatically upload and select the image when somebody clicks the insert button?

If not I will submit a pull request for that.

**Comment by frapell** [comment]
The only reason I can imagine is if you fat fingered the image and select an incorrect one.

If you take Dropbox for example, it will start uploading right away after choosing the image, but if you chose an incorrect one, you can remove it in few clicks.

If you upload an incorrect image using the "Insert image" from tinymce when editing a document, it means you will need to stop editing, go to the image and remove it.

I think in this case, it is ok to have to manually click upload after choosing the image. It would be great if the UI team can come up with a design for users to intuitively understand this

**Comment by vangheem** [comment]
For me on coredev, it *is* still automatically selecting the image you just uploaded. It just doesn't flip back to the right tab automatically for you.

I'm indifferent about automatically uploading immediately after you select a file.

We probably should try having a robot test for this...

**Comment by domruf** [comment]
@vangheem you are right. The image was selected but since it was not visible in the dialog it was a bit confusing.

My idea was that the upload would start after clicking the insert button,
not directly after selecting/dropping a file. That I think would be a bit premature.

**Comment by jensens** [comment]
Closing as outdated. Mockup was completely rewritten for Plone 6 (jQuery/Backbone/RequireJS → ES6/Patternslib). Old Insert Image upload tab wording.
If you think this issue is still relevant in the new implementation, please reopen it. Thank you for the original report!
